### PR TITLE
executor: fix the show variable wrong case sensitive issue#32851

### LINKED ISSUE #32851: TiDB doesn't give any result for upper case sql_mode on SHOW SESSION VARIABLES LIKE `'SQL_MODE'`
### 1. Minimal reproduce step 
```sql
SET @@SQL_MODE='NO_BACKSLASH_ESCAPES';
SHOW SESSION VARIABLES LIKE 'SQL_MODE';
SHOW SESSION VARIABLES LIKE 'sql_mode';
```

### 2. What did you expect to see? 
```sql
mysql> SET @@SQL_MODE='NO_BACKSLASH_ESCAPES';
Query OK, 0 rows affected (0.00 sec)

mysql> SHOW SESSION VARIABLES LIKE 'SQL_MODE';
+---------------+----------------------+
| Variable_name | Value                |
+---------------+----------------------+
| sql_mode      | NO_BACKSLASH_ESCAPES |
+---------------+----------------------+
1 row in set (0.01 sec)

mysql> SHOW SESSION VARIABLES LIKE 'sql_mode';
+---------------+----------------------+
| Variable_name | Value                |
+---------------+----------------------+
| sql_mode      | NO_BACKSLASH_ESCAPES |
+---------------+----------------------+
1 row in set (0.00 sec)
```

### 3. What did you see instead
```sql
tidb> SET @@SQL_MODE='NO_BACKSLASH_ESCAPES';
Query OK, 0 rows affected (0.00 sec)

tidb> SHOW SESSION VARIABLES LIKE 'SQL_MODE';   -- doesn't show any result
Empty set (0.01 sec)

tidb> SHOW SESSION VARIABLES LIKE 'sql_mode';
+---------------+----------------------+
| Variable_name | Value                |
+---------------+----------------------+
| sql_mode      | NO_BACKSLASH_ESCAPES |
+---------------+----------------------+
1 row in set (0.01 sec)
```

### 4. What is your TiDB version? 
```
tidb_version(): Release Version: v5.5.0-alpha-210-g11f4ca802
Edition: Community
Git Commit Hash: 11f4ca802083ee38d5972730ba8f9b72395316fb
Git Branch: master
UTC Build Time: 2022-02-24 04:52:01
GoVersion: go1.17.2
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
1 row in set (0.02 sec)
```

### PR DESCRIPTION
**Pull request by fanrenhoo**
Signed-off-by: fanrenhoo <fanrenhoo@sina.com>

### What problem does this PR solve?

Issue Number: close #32851

Problem Summary: show session variable should not be case sensitive

### What is changed and how it works?
show session variable should not be case sensitive, so make the pattern like show predicate extract to handle the like pattern string compare be case insensitive like show tables and show columns

I tested now it works fine

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
mysql> SET @@SQL_MODE='NO_BACKSLASH_ESCAPES';
mysql> SHOW SESSION VARIABLES LIKE 'SQL_MODE';
+---------------+----------------------+
| Variable_name | Value                |
+---------------+----------------------+
| sql_mode      | NO_BACKSLASH_ESCAPES |
+---------------+----------------------+
1 row in set (0.01 sec)

mysql> SHOW SESSION VARIABLES LIKE 'sql_mode';
+---------------+----------------------+
| Variable_name | Value                |
+---------------+----------------------+
| sql_mode      | NO_BACKSLASH_ESCAPES |
+---------------+----------------------+
1 row in set (0.00 sec)

- [ ] No code

Side effects
no
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation
no
- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
Make the SHOW SESSION VARIABLE LIKE statement case insensitive. before the SHOW SESSION VARIABLE LIKE is case sensitive, not compatible with mysql, now it fixed.
```

### CODE REVIEW

File: planner/core/show_predicate_extractor.go [hunk 1]
```diff
@@ -138,3 +139,46 @@ func (e *ShowTablesTableExtractor) explainInfo() string {
 	}
 	return s
 }
+
+// ShowVariablesExtractor is used to extract some predicates of variables.
+type ShowVariablesExtractor struct {
+	ShowColumnsTableExtractor
```
Comment (hawkingrei) [inline planner/core/show_predicate_extractor.go:145]: ```Extract``` is the same one as the ```ShowTablesTableExtractor```. Can we Inherit it from the parent class? this is that parent class we create a new one and call it ```BaseExtractor```.

Comment (fanrenhoo) [reply planner/core/show_predicate_extractor.go:145]: OK,  I'll learn and check, THKS.

Comment (fanrenhoo) [reply planner/core/show_predicate_extractor.go]: I made changes to be unified ShowBaseExtrator format, please check, THKS much.

**Review by hawkingrei — APPROVED** [review]

**Comment by hawkingrei** [comment]
@fanrenhoo  Please fix the linter.

**Review by xhebox — APPROVED** [review]

**Pull request merged**